src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -514,13 +514,13 @@ private static boolean startsWithInheritDoc(DetailNode root) {
         boolean found = false;
         final DetailNode[] children = root.getChildren();
 
-        for (int i = 0; !found; i++) {
-            final DetailNode child = children[i];
+        for (DetailNode child : children) {
```

Comment:
` for (DetailNode child : root.getChildren()) {`

Reply:
done